relayer/relayer.go

```diff
@@ -0,0 +1,165 @@
+package relayer
+
+import (
+	"fmt"
+	"math/big"
+	"net"
+	"path"
+	"strconv"
+	"strings"
+
+	"github.com/0xPolygon/polygon-edge/contracts"
+	"github.com/0xPolygon/polygon-edge/tracker"
+	"github.com/0xPolygon/polygon-edge/txrelayer"
+	"github.com/0xPolygon/polygon-edge/types"
+
+	hcf "github.com/hashicorp/go-hclog"
+	"github.com/umbracle/ethgo"
+	"github.com/umbracle/ethgo/abi"
+	"github.com/umbracle/ethgo/jsonrpc"
+)
+
+var commitEvent = abi.MustNewEvent(`event NewBundleCommit(uint256 startId, uint256 endId, bytes32 root)`)
+
+type Relayer struct {
+	dataDir           string
+	rpcEndpoint       string
+	stateReceiverAddr ethgo.Address
+	logger            hcf.Logger
+	client            *jsonrpc.Client
+	txRelayer         txrelayer.TxRelayer
+	key               ethgo.Key
+}
+
+func sanitizeRPCEndpoint(rpcEndpoint string) string {
+	if rpcEndpoint == "" || strings.Contains(rpcEndpoint, "0.0.0.0") {
+		_, port, err := net.SplitHostPort(rpcEndpoint)
+		if err == nil {
+			rpcEndpoint = fmt.Sprintf("http://%s:%s", "127.0.0.1", port)
+		} else {
+			rpcEndpoint = "http://127.0.0.1:8545"
+		}
+	}
+
+	return rpcEndpoint
+}
+
+func NewRelayer(
+	dataDir string,
+	rpcEndpoint string,
+	stateReceiverAddr ethgo.Address,
+	logger hcf.Logger,
+	key ethgo.Key,
+) *Relayer {
+	endpoint := sanitizeRPCEndpoint(rpcEndpoint)
+
+	// create the JSON RPC client
+	client, err := jsonrpc.NewClient(endpoint)
+	if err != nil {
+		logger.Error("Failed to create the JSON RPC client", "err", err)
+
+		return nil
+	}
+
+	txRelayer, err := txrelayer.NewTxRelayer(txrelayer.WithClient(client))
```

Comment:
Could you add error log there?

Reply:
🦅 👁️ https://github.com/0xPolygon/polygon-edge/pull/1044/commits/3b6b5da9392e5895c8602434463edbd621220847